cli/app.go

```diff
@@ -2500,6 +2500,38 @@ Note: There is no progress meter while copying is in progress.
 				},
 			},
 		},
+		{
+			Name:            "metadata",
+			Usage:           "manage the metadata attached to your orgs, locations, machines, and/or machine parts",
+			UsageText:       createUsageText("metadata", nil, false, true),
+			HideHelpCommand: true,
+			Subcommands: []*cli.Command{
+				{
+					Name:      "read",
+					Usage:     "read metadata attached to your orgs, locations, machines, and/or machine parts",
+					UsageText: "Provide at least one of the identifiers using CLI arguments to fetch the corresponding metadata",
+					Flags: []cli.Flag{
+						&cli.StringFlag{
+							Name:  "organization-id",
+							Usage: "ID of the organization you want to read the metadata from",
+						},
+						&cli.StringFlag{
+							Name:  "location-id",
+							Usage: "ID of the location you want to read the metadata from",
+						},
+						&cli.StringFlag{
+							Name:  "machine-id",
+							Usage: "ID of the machine you want to read the metadata from",
+						},
+						&cli.StringFlag{
+							Name:  "machine-part-id",
+							Usage: "ID of the machine part you want to read the metadata from",
+						},
+					},
```

Comment:
Is it true that these have to be IDs? Is passing names not sufficient?

If this works with names we should use `commonPartFlags` instead of recreating the wheel here. Even if not, prefer using existing flag names rather than redefining commonly used ones here.

Reply:
ahh actually `commonPartFlags` might not be quite right because it lists `part` as required (which it wouldn't be here). Even so, I think we can pretty trivially DRY this up a bit.